homeassistant/components/sensor/broadlink.py

```diff
@@ -117,12 +117,14 @@ def __init__(self, interval, ip_addr, mac_addr, timeout):
     def _update(self, retry=3):
         try:
             data = self._device.check_sensors_raw()
-            if (data is not None and data.get('humidity', 0) <= 100 and
-                    data.get('light', 0) in [0, 1, 2, 3] and
-                    data.get('air_quality', 0) in [0, 1, 2, 3] and
-                    data.get('noise', 0) in [0, 1, 2]):
-                self.data = data
-                return
+            if data is not None:
+                if data.get('temperature', 0) <= 100 and
```

Comment:
This is not a very elegante solution.
Disable the test or do a restructure as you suggested:
>> Eventually I agree with the comment in the other PR, we should add some more structured validation to the returned data, but for now, this'll do.